### PR TITLE
ci: fix workflow triggers for forks

### PR DESCRIPTION
**Pull request by matejchalk**
PR from forks weren't triggering our main CI workflow, and other workflows were broken. Attempted to fix these problems and tested what I could.
- CI workflow now runs on `pull_request` event also (forks don't run on `push`).
- Code PushUp workflow needs access to secrets, so it is now skipped for forks. 
- Create a separate Code PushUp workflow without secrets that should run only for forks. It won't upload anything, but the PR comments should still work. Can't test it before merging the workflow though, due to [how `pull_request_target` event works](https://github.com/orgs/community/discussions/54937#discussioncomment-5848312).
- Assigning PR author failed for my fork because of invalid permissions, should be solved by changing `pull_request` to `pull_request_target` (again, can't test until it's merged).

Tested triggers on:
- #742

<details><summary> :framed_picture:  tested PR from fork</summary>
<p>

image

</p>
</details>

### CODE REVIEW

**Review by vmasek — DISMISSED** [review]

**Comment by vmasek** [comment]
Just to remember there is also this setting, I guess the approve first setting is correct
image

**Review by BioPhoton — COMMENTED** [review]
Nice!! 

Should we add a small document explaining the if? or add docs to the yml directly?

File: .github/workflows/code-pushup-fork.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Code PushUp (fork)
+
+# separated from code-pushup.yml for security reasons
+# => requires permissions to create PR comment
+# => for PRs from forks, needs to run on `pull_request_target`, not `pull_request`
+# => `pull_request_target` is a security risk when secrets are being used
+# => secrets needed for code-pushup upload
+# => code-pushup for forks runs in separate workflow with no secrets access
```
Comment (BioPhoton) [inline .github/workflows/code-pushup-fork.yml:8]: Thanks!

**Review by BioPhoton — APPROVED** [review]
Thanks for adding a comment!

**Pull request merged**